src/internal/packager/deploy.go

```diff
@@ -251,12 +241,31 @@ func deployComponents(tempPath tempPaths, component types.ZarfComponent) {
 	}
 
 	if hasImages {
-		images.PushToZarfRegistry(tempPath.images, component.Images)
+		// Try image push up to 3 times
+		for retry := 0; retry < 3; retry++ {
```

Comment:
Now that we're adding retries into the image & repo pushes, we should consider doing the pushes in a Goroutine. That seams like the ideal use case for a Goroutine.

We would just need to make sure all the routines are finished by the time we start applying the `component.Charts`

Reply:
Not asking this to be done in this PR. More so trying to see if you agree so I can create an issue in the backlog.

Reply:
Yeah I thought about that too, agree it needs to at least be in a separate issue--but I do have some concerns.  I know w #454 was talking about increasing CPU limits for gitea because they were parallelizing via a script for the git repos and saturate CPU. We already saturate CPU for the registry when pushing sequentially so think we need to look at that carefully.  Also note that a runner is [only 2 cores](https://docs.github.com/en/actions/using-github-hosted-runners/about-github-hosted-runners#supported-runners-and-hardware-resources)--so we run out of CPU real fast if we try to get fancy with autoscaling.